### PR TITLE
Series.values deprecation of converts and drops

### LINKED ISSUE #54717: DEPR: values casting dt64tz, Interval, Period
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
NA
```

### Issue Description

ATM series.values drops the tz from dt64tz, converts interval to object, converts period to object, but preserves other EAs.  We have kept this weird behavior for backward compatibility, should eventually just bite the bullet and deprecate these weird behaviors, just match `_values`

### Expected Behavior

NA

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by pedrocariellof**
- [x] closes #54717 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Added the warnings in the function external_values in "pandas/core/internals/blocks.py" and added most of the asserts in the tests. Some tests will not pass, because they sometimes would show the warning and sometimes wouldn't so I didn't know what to do.

### CODE REVIEW

**Comment by pedrocariellof** [comment]
I was thinking about changing the stack_level on the warning so we don't need to do as much asserts on the tests.

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -2349,11 +2349,21 @@ def external_values(values: ArrayLike) -> ArrayLike:
     proper extension array).
     """
     if isinstance(values, (PeriodArray, IntervalArray)):
+        warnings.warn(
+            "series.values will stop converting tz from dt64tz, interval to object and period to object",
+            FutureWarning,
+            stacklevel=find_stack_level(),
+        )
```
Comment (rhshadrach) [inline pandas/core/internals/blocks.py:2356]: Can you make this warning specific to period / interval. In other words, if they have period data, they should just see the message for period.

File: pandas/core/internals/blocks.py [hunk 2]
```diff
@@ -2349,11 +2349,21 @@ def external_values(values: ArrayLike) -> ArrayLike:
...
         return values.astype(object)
     elif isinstance(values, (DatetimeArray, TimedeltaArray)):
         # NB: for datetime64tz this is different from np.asarray(values), since
         #  that returns an object-dtype ndarray of Timestamps.
         # Avoid raising in .astype in casting from dt64tz to dt64
+        warnings.warn(
+            "series.values will stop converting tz from dt64tz, interval to object and period to object",
+            FutureWarning,
+            stacklevel=find_stack_level(),
+        )
```
Comment (rhshadrach) [inline pandas/core/internals/blocks.py:2366]: If the data is tz-agnostic, then there will be no behavior change. So users should not see a warning.

File: pandas/tests/computation/test_eval.py [hunk 3]
```diff
@@ -1507,12 +1507,16 @@ def test_nested_period_index_subscript_expression(self):
...
+        with tm.assert_produces_warning(
+            FutureWarning,
+            match="series.values will stop converting tz from dt64tz, interval to object and period to object",
+        ):
+            res = self.eval(
+                "df.dates1 < 20130101",
+                local_dict={"df": df},
+                engine=engine,
+                parser=parser,
+            )
```
Comment (rhshadrach) [inline pandas/tests/computation/test_eval.py:1519]: The "user" is not using `.values` here, they should not see a warning (at least for .values).

**Review by rhshadrach — COMMENTED** [review]
Thanks for the PR! In many of the tests, users would be getting a warning about `.values` even when they are not using `.values`. We need to:

 - Determine when the behavior will change on the user, and only raise warnings when it will.
 - For cases that will change, raise an appropriate error message. It should make sense to the user why they are getting the warning, and how they can work around it to either preserve the old behavior or adopt future behavior (and silence the warning).

File: pandas/core/internals/blocks.py (hunk 2 above)
Comment (pedrocariellof) [reply pandas/core/internals/blocks.py:2366]: How can I see if the data is tz-agnostic?

Comment (rhshadrach) [reply pandas/core/internals/blocks.py:2366]: I believe `isinstance(values.dtype, pd.DatetimeTZDtype)`

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**